util/src/main/java/tc/oc/pgm/util/nms/NMSHacksSportPaper.java

```diff
@@ -0,0 +1,258 @@
+package tc.oc.pgm.util.nms;
+
+import com.google.common.collect.SetMultimap;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import net.minecraft.server.v1_8_R3.DataWatcher;
+import net.minecraft.server.v1_8_R3.PacketPlayOutAttachEntity;
+import net.minecraft.server.v1_8_R3.PacketPlayOutNamedEntitySpawn;
+import net.minecraft.server.v1_8_R3.PacketPlayOutPlayerInfo;
+import net.minecraft.server.v1_8_R3.PacketPlayOutScoreboardTeam;
+import net.minecraft.server.v1_8_R3.PacketPlayOutSpawnEntity;
+import net.minecraft.server.v1_8_R3.PacketPlayOutSpawnEntityLiving;
+import org.bukkit.Bukkit;
+import org.bukkit.ChunkSnapshot;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.BlockState;
+import org.bukkit.craftbukkit.v1_8_R3.inventory.CraftItemStack;
+import org.bukkit.craftbukkit.v1_8_R3.scoreboard.CraftTeam;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.DoubleChestInventory;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.scoreboard.NameTagVisibility;
+import tc.oc.pgm.util.attribute.AttributeModifier;
+import tc.oc.pgm.util.skin.Skin;
+
+public class NMSHacksSportPaper extends NMSHacks1_8 {
+  @Override
+  public void updateChunkSnapshot(ChunkSnapshot snapshot, BlockState blockState) {
+    snapshot.updateBlock(blockState);
+  }
+
+  @Override
+  public void setKnockbackReduction(Player player, float amount) {
+    player.setKnockbackReduction(amount);
+  }
+
+  @Override
+  public void showInvisibles(Player player, boolean showInvisibles) {
+    player.showInvisibles(showInvisibles);
+  }
+
+  @Override
+  public void setAffectsSpawning(Player player, boolean affectsSpawning) {
+    player.spigot().setAffectsSpawning(affectsSpawning);
+  }
+
+  @Override
+  public void entityAttach(Player player, int entityID, int vehicleID, boolean leash) {
+    sendPacket(player, new PacketPlayOutAttachEntity(entityID, vehicleID, leash));
```

Comment:
```suggestion
    NMSHacksProvider.get().sendPacket(player, new PacketPlayOutAttachEntity(entityID, vehicleID, leash));
```
I saw you doing something like this at some other place in your code and I think it should be the standard way of calling methods. In case some NMSHacks instance down the line overrides some other this would call the "most correct" implementation of the method.